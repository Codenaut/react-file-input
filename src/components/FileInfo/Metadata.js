import React from 'react';
import PropTypes from 'prop-types';
import filesize from 'filesize';

import '../../styles/FileInfo.scss';

const Metadata = ({ extension, name, size, type }) => {
  const formattedSize = filesize(size, { separator: ',' });

  return (
    <div className="brainhub-file-info__metadata">
      <div className="brainhub-file-info__metadata__info">
        <span className="brainhub-file-info__metadata__info_name">{name}</span>{extension ? `.${extension}`: ''} ({formattedSize}) {type}
      </div>
    </div>
  );
};

Metadata.defaultProps = {
  extension: '',
  type: '',
};

Metadata.propTypes = {
  extension: PropTypes.string,
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  size: PropTypes.number.isRequired,
};

export default Metadata;

