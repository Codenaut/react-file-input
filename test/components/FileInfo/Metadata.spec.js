import React from 'react';
import renderer from 'react-test-renderer';

import Metadata from '../../../src/components/FileInfo/Metadata';

describe('components', () => {
  describe('FileInfo', () => {
    describe('Thumbnail', () => {
      it('should match exact snapshot', () => {
        const tree = renderer.create(
          <div>
            <Metadata name="test" size={1000}/>
          </div>
        ).toJSON();

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
