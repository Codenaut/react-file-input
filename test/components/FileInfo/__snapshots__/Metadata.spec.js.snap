// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components FileInfo Thumbnail should match exact snapshot 1`] = `
<div>
  <div
    className="brainhub-file-info__metadata"
  >
    <div
      className="brainhub-file-info__metadata__info"
    >
      <span
        className="brainhub-file-info__metadata__info_name"
      >
        test
      </span>
      
       (
      1000 B
      ) 
      
    </div>
  </div>
</div>
`;
